### DISCUSSION TITLE
Prevent default behavior for canvas-release / canvas-drag-end events

### DISCUSSION THREAD

**Discussion by miordache93** [openseadragon/openseadragon#2624] [Q&A]
Is there any way that i can prevent the on click event be fired when user finish dragging the canvas?

Much appreciate it

**Comment by iangilman** [comment]
What are you trying to accomplish? You can set `preventDefaultAction` on the event when you get a `canvas-click`. You can also check the `quick` flag on the event to see if it was a drag vs. a click. 

As far as I know, there's no way to prevent the `canvas-click` event from being fired from within the `canvas-release` or `canvas-drag-end` events. If you really want that, you could set a flag in those events that you could read in the click event. 

Any other thoughts @msalsbery?

**Reply by miordache93** [reply]
Requirement is to add a markup at the position where the user clicks in the canvas, but at the same time user need to be able to drag on the canvas. Therefore the drag event interfere with the markup creation, creating additional markups in the canvas.

PS: For markups i'm using osd overlays with MouseTracker events, but i don't think is relevant.

**Reply by iangilman** [reply]
Cool... sounds like checking the `quick` flag should work for you. If it's a quick click, add a markup. If it's not, don't add one because it was a drag.
